src/Vipps.net/Infrastructure/BaseServiceClient.cs

```diff
@@ -204,5 +225,61 @@ private static void AddOrUpdateHeader(HttpRequestHeaders headers, string key, st
 
             headers.Add(key, value);
         }
+
+        private static HttpContent CreateFormRequestContent<TRequest>(TRequest vippsRequest)
+            where TRequest : class
+        {
+            if (vippsRequest is null)
+            {
+                return null;
+            }
+
+            var keyValue = ToKeyValue(vippsRequest);
+            return new FormUrlEncodedContent(keyValue);
+        }
+
+        public static IDictionary<string, string> ToKeyValue(object metaToken)
+        {
+            if (metaToken == null)
+            {
+                return null;
+            }
+
+            JToken token = metaToken as JToken;
+            if (token == null)
+            {
+                return ToKeyValue(JObject.FromObject(metaToken));
+            }
+
+            if (token.HasValues)
+            {
+                var contentData = new Dictionary<string, string>();
+                foreach (var child in token.Children().ToList())
+                {
+                    var childContent = ToKeyValue(child);
+                    if (childContent != null)
+                    {
+                        contentData = contentData
+                            .Concat(childContent)
+                            .ToDictionary(k => k.Key, v => v.Value);
+                    }
+                }
+
+                return contentData;
+            }
+
+            var jValue = token as JValue;
+            if (jValue?.Value == null)
+            {
+                return null;
+            }
+
+            var value =
+                jValue?.Type == JTokenType.Date
+                    ? jValue?.ToString("o", CultureInfo.InvariantCulture)
+                    : jValue?.ToString(CultureInfo.InvariantCulture);
+
+            return new Dictionary<string, string> { { token.Path, value } };
+        }
```

Comment:
And all these lines of code were the reason I looked for the usage of the method above